### ISSUE TITLE
Support newer versions of momentjs etc.

### ISSUE THREAD

**Issue by supersteves** [handsontable/handsontable#5159]
Your `package.json` file specifies 3 deps:
```
    "moment": "2.20.1",
    "numbro": "1.11.0",
    "pikaday": "1.5.1"
```
Now I'd like to use the latest momentjs (2.22.2) in my project (since I already use momentjs). I can't without having two copies of moment bundled at runtime - the one I use (today's version) and the handsontable dep (from last December).

Can you change your package.json to have something like:
```
    "moment": "^2.20.1",
    "numbro": "^1.11.0",
    "pikaday": "^1.5.1"
```
i.e. "2.20.1 or later" etc.?

**Comment by swrobel** [comment]
I attempted to make this change previously. I don't know why it's so difficult to get them to accept it: https://github.com/handsontable/handsontable/pull/5047

**Comment by mismith** [comment]
Where is this at now @swrobel?

**Comment by swrobel** [comment]
@mismith 🤷🏼‍♂️

**Comment by wojciechczerniak** [comment]
Dependency pinning/freezing/locking is a valid and responsible approach when you're shipping a product. We want to be sure that our software is tested with dependencies that we rely on. And sorry, but we cannot test them all.

Providing a version range is also a form of dependency locking. You have to have a confidence in your vendors or transfer the responsibility for testing the product to your users. Great if you have a side project for fun that is shared "as is, without warranty of any kind" (quote from MIT). Otherwise, things get a little complicated.

We're going to rethink dependencies when we move to Lerna.js. Splitting Handsontable should help in having finer control over final app build. 

---
BTW Can I ask you all why you're compiling two versions of moment if you don't want to? You're in control of your build scripts and their output. Few ideas how you can do it:

**Yarn** [resolutions](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/):
```json
"resolutions": {
    "handsontable/moment": "^2.23.0"
}
```

**Webpack** [resolve.alias](https://webpack.js.org/configuration/resolve/#resolve-alias)
```js
"resolve": {
    "alias": {
        "moment": path.resolve(__dirname, 'node_modules/moment'),
    }
}
```

**CDN** (use the `handsontable.js` build, without deps)
```html
<script src="https://cdn.jsdelivr.net/npm/pikaday@latest/pikaday.js"></script>
<script src="https://cdn.jsdelivr.net/npm/numbro@latest/dist/numbro.js"></script>
<script src="https://cdn.jsdelivr.net/npm/moment@latest/moment.js"></script>
<script src="https://cdn.jsdelivr.net/npm/handsontable/dist/handsontable.js"></script>
```
---
So both requirements are met currently. We're confident that everything is tested with the dependencies versions we provide. And anyone who want to experiment can still do it. Win-Win.

Of course any improvement is welcome. We're open, we just have to do it carefully.

**Comment by chagar** [comment]
pikaday 1.5.1 is over two years old. This is bound to cause compatibility problems with up-to-date moment, webpack, babel, etc. The problem is not merely the use of pinning, but pinning ancient versions.

**Comment by AMBudnik** [comment]
Hey @chagar we're waiting for learna to be developed. It will allow us to stay up to date with all dependencies.

**Comment by AMBudnik** [comment]
Hi @supersteves @chagar @camsong @gnewtzie @swrobel @mismith @vkrol 

we've just merged https://github.com/handsontable/handsontable/pull/6610 which 

> Bump dependencies:
>
> moment to fixed v2.24.0;
> numbro to fixed v2.1.2;
> pikaday to fixed v1.8.0;

into Handsontable 8.0.0-beta2


Here’s a full list of changes https://github.com/handsontable/handsontable/releases/tag/8.0.0-beta.2
NPM https://www.npmjs.com/package/handsontable/v/8.0.0-beta.2 

All feedback's appreciated. Thank you for your input.

**Comment by AMBudnik** [comment]
Hi @supersteves @chagar @gnewtzie @swrobel @mismith

We've just released v 8.0.0 (stable) that provides the mentioned changes

```
moment to fixed v2.24.0;
numbro to fixed v2.1.2;
pikaday to fixed v1.8.0;
```

Related topic https://github.com/handsontable/handsontable/pull/6610
